sonic_platform_base/psu_base.py

```diff
@@ -0,0 +1,99 @@
+#
+# psu_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a power supply unit (PSU) in SONiC
+#
+
+try:
+    import abc
+    from . import device_base
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+
+class PsuBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a power supply unit
+    """
+    __metaclass__ = abc.ABCMeta
+
+    # Possible fan directions
+    FAN_DIRECTION_INTAKE = "intake"
```

Comment:
sounds good to me to have PSU derived from FAN base.

Reply:
@keboliu: My suggestion was to have a PSU object contain one fan object, rather than derive PsuBase from FanBase. I think this makes more sense, especially if certain PSU modules contain more than fan. Do you agree? Also, are you aware of any PSU module which contains more than one fan?